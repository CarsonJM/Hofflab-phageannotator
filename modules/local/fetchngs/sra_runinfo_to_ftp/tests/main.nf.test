nextflow_process {

    name "Test process: SRA_RUNINFO_TO_FTP"
    script "../main.nf"
    process "SRA_RUNINFO_TO_FTP"

    test("SRR13191702.runinfo.tsv") {

        when {
            process {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'tsv/SRR13191702.runinfo.tsv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("SRR13191702.runinfo.tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = file(params.pipelines_testdata_base_path + 'tsv/SRR13191702.runinfo.tsv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
