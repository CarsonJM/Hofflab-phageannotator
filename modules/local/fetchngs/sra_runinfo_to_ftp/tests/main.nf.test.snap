{
    "SRR13191702.runinfo.tsv": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e"
                ],
                "1": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ],
                "tsv": [
                    "SRR13191702.runinfo_ftp.tsv:md5,94378c448c044b3e20e5c54e442ab62e"
                ],
                "versions": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-22T11:07:48.234610233"
    },
    "SRR13191702.runinfo.tsv - stub": {
        "content": [
            {
                "0": [
                    "SRR13191702.runinfo_ftp.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ],
                "tsv": [
                    "SRR13191702.runinfo_ftp.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,e95f8185f665127a73622a19d321bcca"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-22T11:07:59.555529627"
    }
}
